assets/app/scripts/services/data.js

```diff
@@ -605,15 +613,21 @@ angular.module('openshiftConsole')
       protocol = window.location.protocol === "http:" ? "http" : "https"; 
     }
 
+    var namespaceInPath = params.namespace && SERVER_TYPE_MAP[type].namespacePath;
+    var namespace = null;
+    if (namespaceInPath) {
+      namespace = params.namespace;
```

Comment:
How does the URL template behave if we get in here and namespace ends up null or undefined? Should we default to ""?